client/plots/disco/Disco.ts

```diff
@@ -15,23 +15,28 @@ import { RingType } from './ring/RingType.ts'
 import Settings from './Settings.ts'
 
 export default class Disco {
-	private type: string
 	private discoInteractions: DiscoInteractions
-	private opts: any
 
+	// following attributes are required by rx
+	private type: string
+	private opts: any
 	private state: any
 	private id: any
 	private app: any
 
 	constructor(opts: any) {
 		this.type = 'Disco'
 		this.opts = opts
-		this.discoInteractions = new DiscoInteractions(this)
```

Comment:
@aacic this line is moved to main(), as DiscoInteractions requires this.state. please verify this change is okay

Reply:
It's fine. It's not super expensive to create the DiscoInteractions object in main.